File: FindMyFamCore/Sources/FindMyFamCore/Models/NostrIdentity.swift
import Foundation

/// The user's Nostr identity — public-facing only.
/// The private key (nsec) lives exclusively in the Keychain, accessed via IdentityService.
public struct NostrIdentity: Equatable {
    /// Full bech32-encoded public key: "npub1..."
    public let npub: String

    /// Raw hex public key (for internal Nostr event authoring)
    public let publicKeyHex: String

    public init(npub: String, publicKeyHex: String) {
        self.npub = npub
        self.publicKeyHex = publicKeyHex
    }

    /// Abbreviated form for UI display: "npub1abc...xyz"
    public var shortNpub: String {
        guard npub.count > 16 else { return npub }
        return "\(npub.prefix(10))...\(npub.suffix(6))"
    }
}

File: android/shared/src/main/java/org/findmyfam/shared/models/NostrIdentity.kt
package org.findmyfam.shared.models

/**
 * The user's Nostr identity — public-facing only.
 * The private key (nsec) lives exclusively in secure storage, accessed via IdentityService.
 */
data class NostrIdentity(
    /** Full bech32-encoded public key: "npub1..." */
    val npub: String,

    /** Raw hex public key (for internal Nostr event authoring). */
    val publicKeyHex: String
) {
    /** Abbreviated form for UI display: "npub1abc...xyz" */
    val shortNpub: String
        get() = if (npub.length > 16) "${npub.take(10)}...${npub.takeLast(6)}" else npub
}
